karabo_data/reader.py

```diff
@@ -1134,17 +1134,21 @@ def stack_detector_data(train, data, axis=-3, modules=16, only='', xcept=()):
         shapes.add(array.shape)
         modno_arrays[modno] = array
 
+    required_shape = (sorted(shapes, key=lambda x: x[0], reverse = True))[0]
     if len(dtypes) > 1:
         raise ValueError("Arrays have mismatched dtypes: {}".format(dtypes))
     if len(shapes) > 1:
-        raise ValueError("Arrays have mismatched shapes: {}".format(shapes))
+        for modno, array in modno_arrays.items():
+            if array.shape != required_shape:
+                modno_arrays[modno] = np.full(required_shape, np.nan, dtype = array.dtype)
```

Comment:
You don't need to create a new array for all misshaped modules as an array of the right size is already made, line 1152